planner/core/task.go

```diff
@@ -106,14 +113,19 @@ func attachPlan2Task(p PhysicalPlan, t task) task {
 
 // finishIndexPlan means we no longer add plan to index plan, and compute the network cost for it.
 func (t *copTask) finishIndexPlan() {
-	if !t.indexPlanFinished {
-		t.cst += t.count() * netWorkFactor
-		t.indexPlanFinished = true
-		if t.tablePlan != nil {
-			t.tablePlan.(*PhysicalTableScan).stats = t.indexPlan.statsInfo()
-			t.cst += t.count() * scanFactor
-		}
+	if t.indexPlanFinished {
+		return
+	}
+	cnt := t.count()
+	t.indexPlanFinished = true
+	// Network cost of transferring rows of index scan to TiDB.
+	t.cst += cnt * netWorkFactor * t.tableStats.HistColl.GetAvgRowSize(t.indexPlan.Schema().Columns, true)
+	if t.tablePlan == nil {
+		return
 	}
+	t.tablePlan.(*PhysicalTableScan).stats = t.indexPlan.statsInfo()
```

Comment:
It's better to add a comment about why we calculating the cost for table scan on coprocessor when we finish the index plan of the cop task.